CONTRIBUTING.md

```diff
@@ -1,3 +1,178 @@
+# Contributing to OpenTelemetry PHP
+
+## Introduction
+
+Welcome to the OpenTelemetry PHP repository! 🎉
+
+Thank you for considering contributing to this project. Whether you're fixing a bug, adding new features, improving documentation, or reporting an issue, we appreciate your help in making OpenTelemetry better. 
+
+This repository is part of the OpenTelemetry ecosystem, which provides observability tooling for distributed systems. Your contributions help enhance the PHP ecosystem for OpenTelemetry users worldwide.
+
+If you have any questions, feel free to ask in the community channels. We’re happy to help! 😊
+
+## Pre-requisites
+
+Before getting started, ensure you have the following installed:
+
+- **PHP** (8.1 or higher) – [Install PHP](https://www.php.net/downloads)
+- **Composer** – [Install Composer](https://getcomposer.org/)
+- **Docker & Docker Compose** – [Install Docker](https://docs.docker.com/engine/install/)
+- **Make** (for running development tasks)
+
+Additional Notes:
+- Windows users may need [Git Bash](https://gitforwindows.org/) for better compatibility.
+- Consider using [phpenv](https://github.com/phpenv/phpenv) for managing multiple PHP versions.
+
+## Workflow
+
+We follow a structured workflow to ensure smooth collaboration:
+
+### Branch Naming Convention
+- **Feature branches**: `feature/<short-description>`
+- **Bugfix branches**: `fix/<short-description>`
+- **Documentation updates**: `docs/<short-description>`
+
+### Commit Message Format
+- Use descriptive commit messages (e.g., `fix(tracing): resolve issue with span context`)
+- Follow [Conventional Commits](https://www.conventionalcommits.org/) where possible.
+
+### Pull Request Guidelines
+- Fork the repository and create a new branch.
+- Follow the coding guidelines before submitting your PR.
+- Ensure tests pass locally before pushing.
+- Link relevant issues in the PR description.
+
+
+## Local Run/Build
+
+To set up your local development environment:
+
+```bash
+# Clone the repository
+git clone https://github.com/open-telemetry/opentelemetry-php.git
+cd opentelemetry-php
+
+# Install dependencies
+make install
+```
+
+To update dependencies:
+
+```bash
+make update
+```
+
+To run checks:
+
+```bash
+make all-checks
+```
+
+
+## Testing
+
+We use various tools for testing and static analysis:
+
+```bash
```

Comment:
I think it's worth mentioning that `make all` is the least you need to know as a new contributor, and covers everything that github actions will check.